test/integrations/stalltracking.test.ts

```diff
@@ -0,0 +1,333 @@
+import { Span, Transaction } from "@sentry/tracing";
+
+import { StallTracking } from "../../src/js/integrations";
+
+describe("StallTracking", () => {
+  it("Stall tracking detects a JS stall", async () => {
+    const stallTracking = new StallTracking();
+
+    const transaction = new Transaction({
+      name: "Test Transaction",
+    });
+    transaction.initSpanRecorder();
+
+    const finishTracking = stallTracking.registerTransactionStart(transaction);
+
+    // We create a very **nasty** object
+    const nastyObject: { value: string[] } = { value: [] };
+    for (let i = 0; i < 100000; i++) {
+      nastyObject.value.push("nasty");
+    }
+
+    // This works in sync, so it should stall the js event loop
+    for (let i = 0; i < 50; i++) {
+      JSON.parse(JSON.stringify(nastyObject));
+    }
+
+    await new Promise<void>((resolve) => {
+      setTimeout(() => {
+        const measurements = finishTracking();
+
+        expect(measurements).not.toEqual(null);
+        if (measurements !== null) {
+          expect(measurements.stall_count.value).toBeGreaterThan(0);
+          expect(measurements.stall_longest_time.value).toBeGreaterThan(0);
+          expect(measurements.stall_total_time.value).toBeGreaterThan(0);
+        }
+
+        resolve();
+      }, 500);
+    });
+  });
+
+  it("Stall tracking timeout is stopped after finishing all transactions (single)", () => {
+    const stallTracking = new StallTracking();
+
+    const transaction = new Transaction({
+      name: "Test Transaction",
+    });
+
+    const finishTracking = stallTracking.registerTransactionStart(transaction);
+
+    const measurements = finishTracking();
+
+    expect(measurements).not.toBe(null);
+
+    // If the stall tracking does not correctly stop, the process will keep running. We detect this by passing --detectOpenHandles to jest.
```

Comment:
Can we use jest's `done()` callback here?

Reply:
Oh yeah I somehow forgot all about `done()` 😅